source/internal/_cloneWithDescriptors.js

```diff
@@ -0,0 +1,22 @@
+
+
+/**
+ * Performs a shallow clone of an object while preserving property descriptors.
+ *
+ * @private
+ * @param {Object} obj The object to clone
+ * @return {Object} The cloned object
+ */
+export default function _cloneWithDescriptors(obj) {
+  var result = {};
```

Comment:
If we're going to do this, would it make sense to start with the source object's prototype and get rid of our annoying "flattens prototype properties onto the new object" while we're at it?